### PR TITLE
Add test to verify GRMHD analytic solutions

### PR DESCRIPTION
**Pull request by kidder**
The test computes numerical time and spatial derivatives to check that
an analytic solution satisfies the GRMHD equations within a given tolerance.

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### CODE REVIEW

**Review by carmaza — DISMISSED** [review]
Looks good to me. 

Will this be a required feature of every system? If so, there might be things in this code that we can factor out. Maybe something to do in the future.

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_BondiMichel.cpp
Comment (mar-celine) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_BondiMichel.cpp:126]: I assume that I will have to change this away from 1.0 once I've fixed BondiMichel to test that BondiMichel is a solution even for M!=1.0

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_KomissarovShock.cpp
Comment (mar-celine) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_KomissarovShock.cpp:204]: Is the domain particularly dependent on the solution being tested? If not I would factor it into the verify_grmhd_solution code.

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp
Comment (mar-celine) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:42]: Is there a reason why `time` isn't here?

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp
Comment (mar-celine) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:100]: Where are these numbers coming from?

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_AlfvenWave.cpp
Comment (mar-celine) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_AlfvenWave.cpp:170]: It looks like the B fields here don't have a y-component. I think in general the solution tested should be in full 3D.

**Comment by kidder** [comment]
@carmaza yes there should be similar tests for each system and I am sure there are pieces of this that could be generalized to do so

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_BondiMichel.cpp
Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_BondiMichel.cpp:126]: yes, this test would currently fail if you change the mass

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_KomissarovShock.cpp
Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_KomissarovShock.cpp:204]: yes, you need to choose a domain that is appropriate for the solution.  After this PR is merged, the authors of the individual solutions should update the tests to verify the solution at both random and appropriate interesting points (e.g. on coordinate axes, the origin, etc) where the solution is smooth

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp
Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:100]: this is a sixth-order central FD stencil (I'll add a comment)

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp
Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:42]: it's a sixth-order centered FD stencil  (I'll add a comment)

Comment (wthrowe) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:42]: How hard would it be to modify the sixth-order finite difference function we have in TestHelpers.hpp to be usable here?

Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:42]: It could probably be done, but it would not be easy and I don't think it would be worth the effort

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_AlfvenWave.cpp
Comment (kidder) [reply tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/Test_AlfvenWave.cpp:170]: I agree, but I will leave this to the individual authors of the analytic solutions to update these to test the appropriate cases

**Review by mar-celine — DISMISSED** [review]

**Comment by kidder** [comment]
@nilsdeppe or @wthrowe please review (@carmaza had approved it earlier)

File: tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp
Comment (wthrowe) [inline tests/Unit/PointwiseFunctions/AnalyticSolutions/GrMhd/VerifyGrMhdSolution.hpp:109]: You need to add this file to the list in Doxyfile.in to get this documentation to show up.

**Review by wthrowe — CHANGES_REQUESTED** [review]

**Comment by kidder** [comment]
@wthrowe @gsb76 @carmaza updated so documentation shows up

**Review by wthrowe — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]
LGTM :+1:

**Pull request merged**
